components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/listener/IdentityOauthEventHandler.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.listener;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.base.IdentityRuntimeException;
+import org.wso2.carbon.identity.core.bean.context.MessageContext;
+import org.wso2.carbon.identity.core.handler.InitConfig;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.event.IdentityEventConstants;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
+import org.wso2.carbon.identity.oauth.OAuthUtil;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+/**
+ * This is an event handler listening for some of the core user management operations.
+ */
+public class IdentityOauthEventHandler extends AbstractEventHandler {
+
+    private static final Log log = LogFactory.getLog(IdentityOauthEventHandler.class);
+
+    public String getName() {
+
+        return "identityOauthEventHandler";
+    }
+
+    public String getFriendlyName() {
+
+        return "Identity Oauth Event Handler";
+    }
+
+    @Override
+    public void init(InitConfig configuration) throws IdentityRuntimeException {
+
+        super.init(configuration);
+    }
+
+    @Override
+    public int getPriority(MessageContext messageContext) {
+
+        int priority = super.getPriority(messageContext);
+        if (priority == -1) {
+            priority = 51;
+        }
+        return priority;
+    }
+
+
+    @Override
+    public void handleEvent(Event event) throws IdentityEventException {
+
+        if (IdentityEventConstants.Event.POST_SET_USER_CLAIMS.equals(event.getEventName())) {
```

Comment:
we have to check both POST_SET_USER_CLAIMS and POST_SET_USER_CLAIM events

Reply:
accountlockhandler is listening only to postsetclaims and setting the lock/disable claims https://github.com/wso2-extensions/identity-event-handler-account-lock/blob/master/components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java#L188.

Reply:
Since IdentityOauthEventHandler.java  is listening to POST_SET_USER_CLAIM event annd had the implementation to revokeLockedUsersToken, moved it to IdentityOauthEventListener.java  as suggested